docs/asciidoc/auth/authentication.adoc

```diff
@@ -92,3 +92,28 @@ type Todo {
 
 extend type Todo @auth(rules: [{ isAuthenticated: true }])
 ----
+
+== `allowUnauthenticated`
+In some cases, you may want to allow unauthenticated requests while also having auth-based rules. You can use the `allowUnauthenticated` parameter to avoid throwing an exception if no auth is present in the context.
+
+In the example below, only the publisher can see his blog posts even if it is not published yet. Once the blog post is published, anyone can see it.
+
+WARNING: In case of unauthenticated requests, auth parameters will be replaced by `null` in the cypher query. In this case, `publisher: "$jwt.sub"` will become `publisher: null`; This can lead to unwanted results if the field used for querying is nullable in Neo4J.
+
+[source, graphql]
+----
+type BlogPost {
+    id: ID!
+    publisher: User!
+    published: Boolean!
+}
+
+extend type BlogPost {
+    password: String! @auth(rules: [
+        { operations: [READ], where: { OR: [
+            { publisher: "$jwt.sub" },
+            { published: true },
+        ] }, allowUnauthenticated: true }
+    ])
+}
```

Comment:
These type definitions don't match the description in the paragraph above, at the moment they just have auth on the password field. Additionally, `{ publisher: "$jwt.sub" }` wouldn't work because `publisher` is an object type and not a string type. So it should probably look something like the following:

```suggestion
type BlogPost
    @auth(
        rules: [
            {
                operations: [READ]
                where: { OR: [{ publisher: "$jwt.sub" }, { published: true }] }
                allowUnauthenticated: true
            }
        ]
    ) {
    id: ID!
    publisher: String!
    published: Boolean!
}
```

I must add, that the only reason that the `{ published: true }` is by complete and utter fluke - I'm almost tempted to say that it's a bad idea to put this example in the docs because of this, because it might not work in all cases. I imagine it certainly wouldn't work if there was any recursion down relationship fields.

Reply:
I'm not sure to fully understand what could cause an issue in a recursive relationship, I assume this should work the same as the classic `where` auth rule should work. But I'm probably misunderstanding something, can you please give me an example of where you think this shouldn't work.

Reply:
I can't give a specific example, it's more just the fact that the only reason this kind of thing works at all is completely by accident, so it's completely untested and we have no actual confidence that it works. Based on that, there's a higher chance that there will be some edge cases where it doesn't work, but we can face them if and when they arrive!